_drafts/device-logging.md

```diff
@@ -0,0 +1,295 @@
+
+```markdown
+E bort    : {"Network error": "-55", "RSSI": 61, "SSID": "Home-Guest"}
+```
+
+JSON is the most common format for structured logs, and can be written manually in code fairly quickly. It's also fairly readable compared to the unstructured log, too.
+
+#### Skip Log Reading -- Build Visualizations!
+
+The end state of logging is ideally to never read the raw logs ourselves. Once we can empower computers to read all sorts of logs, we can build the tools to parse and then visualize the information, and save ourselves the effort of piecing together all the information in our head.
+
+<p align="center">
+  <img width="850" src="{% img_url blog-device-logging/logging-wifi.png %}" alt="Wi-Fi connection status over time" />
+</p>
+
+Here, instead of manually tracking each Wi-Fi connection and signal strength log over time, we can create a visualization for it, and have an instant understanding of the timing involved.
+
+<p align="center">
+  <img width="850" src="{% img_url blog-device-logging/logging-package-install.png %}" alt="Package Install over time example" />
+</p>
+
+Or here, instead of scanning the logs to see when and which package was installed, we can automatically parse the logs and create a visualization.
+
+<p align="center">
+  <img width="850" src="{% img_url blog-device-logging/logging-package-install-context.png %}" alt="Package Install contextualized example" />
+</p>
+
+Once we've collected enough data, we can plot all the logs as events on a timeline, which allows us to make correlations instantly that we'd have so much more trouble from just reading.
+
+If this sounds familiar, this is pitch for Memfault! Instead of reading logs, we can acquire a lot of the same data that the log would have produced, and then display it on a browsable timeline in extremely high resolution.
+
+## Conclusion
+
+Logging is a fundamental part of our day to day debugging toolset, and it's hard to imagine it ever going away entirely. But we all agree, there is a lot of noise in our logs, there are really no standards for logging, and there's a lot we could probably improve how we log, if we only had the time. Hopefully I've shared some new or helpful ideas around logging that you can put to use today!
+
+<!-- Interrupt Keep START -->
+{% include newsletter.html %}
+
+{% include submit-pr.html %}
+<!-- Interrupt Keep END -->
+
+{:.no_toc}
+
+## Additional Reading
+
+* [Logging v. instrumentation](https://peter.bourgon.org/blog/2016/02/07/logging-v-instrumentation.html)
+* [Tracing: structured logging, but better in every way](https://andydote.co.uk/2023/09/19/tracing-is-better/)
+
+{:.no_toc}
+
+<!-- prettier-ignore-start -->
+[^meaningfulmessages]: [Log meaningful messages](https://newrelic.com/blog/best-practices/best-log-management-practices#toc-log-meaningful-messages-that-drive-decisions-)
+[^structuredlogging]: [Structured logging](https://newrelic.com/blog/how-to-relic/structured-logging)
```

Comment:
Linking to a semi-competitor, how bold of you.

Reply:
If you're confident in the product you shouldn't care, I guess.